### PR TITLE
Cleanup of Makefile.cvc

### PR DESCRIPTION
**Pull request by themperek**
Enable compiled mode
Cleanup
Update documentation

CVC seems to work on some basic examples.

### CODE REVIEW

File: cocotb/share/makefiles/simulators/Makefile.cvc [hunk 1]
```diff
@@ -49,46 +49,39 @@ ifeq (, $(CMD))
...
-        gdb --args $(SIM_BUILD)/sim.vvp $(SIM_ARGS) $(EXTRA_ARGS) $(PLUSARGS)
-endif
 
+    # Compilation phase
+    $(SIM_BUILD)/cvcsim: $(SIM_BUILD) $(VERILOG_SOURCES) $(CUSTOM_COMPILE_DEPS)
+	    $(CMD) +verbose -o $(SIM_BUILD)/cvcsim $(CVC_ARGS) +loadvpi=$(LIB_DIR)/libcocotbvpi_modelsim:vlog_startup_routines_bootstrap $(COMPILE_ARGS) $(EXTRA_ARGS) $(VERILOG_SOURCES)
+
+    # Execution phase
+    $(COCOTB_RESULTS_FILE): $(SIM_BUILD)/cvcsim $(CUSTOM_SIM_DEPS)
+		-@rm -f $(COCOTB_RESULTS_FILE)
```
Comment (eric-wieser) [inline cocotb/share/makefiles/simulators/Makefile.cvc:74]: Indenting recipes like this seems like a recipe for mistakes. For other reviewers, the indentation here is:
```
    $(COCOTB_RESULTS_FILE): $(SIM_BUILD)/cvcsim $(CUSTOM_SIM_DEPS)
<tab><tab>-@rm -f $(COCOTB_RESULTS_FILE)
```

Comment (themperek) [reply cocotb/share/makefiles/simulators/Makefile.cvc:74]: Anything is fine for me. You prefer no intention for ifdef in this case ?

Comment (eric-wieser) [reply cocotb/share/makefiles/simulators/Makefile.cvc:74]: Yeah, I think so - just leave blank lines around the `if` and `else` to make them easy to spot.

**Comment by cmarqu** [comment]
The typo ``CVC_ITERP`` was in before, but I guess you could correct it to ``CVC_INTERP`` in this PR?

File: cocotb/share/makefiles/simulators/Makefile.cvc [hunk 2]
```diff
@@ -47,48 +47,39 @@ endif
...
 
-# Compilation phase
-$(SIM_BUILD)/sim.vvp: $(SIM_BUILD) $(VERILOG_SOURCES) $(CUSTOM_COMPILE_DEPS)
-	MODULE=$(MODULE) \
-	TESTCASE=$(TESTCASE) TOPLEVEL=$(TOPLEVEL) TOPLEVEL_LANG=$(TOPLEVEL_LANG) COCOTB_SIM=1 \
-        $(CMD) $(CVC_ARGS) +acc+2 -o $(SIM_BUILD)/sim.vvp +define+COCOTB_SIM=1 +loadvpi=$(LIB_DIR)/libcocotbvpi_modelsim:vlog_startup_routines_bootstrap $(COMPILE_ARGS) $(EXTRA_ARGS) $(VERILOG_SOURCES)
+	MODULE=$(MODULE) TESTCASE=$(TESTCASE) TOPLEVEL=$(TOPLEVEL) TOPLEVEL_LANG=$(TOPLEVEL_LANG) COCOTB_SIM=1 \
+	$(CMD) +loadvpi=$(LIB_DIR)/libcocotbvpi_modelsim:vlog_startup_routines_bootstrap $(COMPILE_ARGS) $(SIM_ARGS) $(EXTRA_ARGS) $(VERILOG_SOURCES) $(PLUSARGS)
+
+	$(call check_for_results_`file)
```
Comment (eric-wieser) [inline cocotb/share/makefiles/simulators/Makefile.cvc:62]:
```suggestion
	$(call check_for_results_file)
```

File: cocotb/share/makefiles/simulators/Makefile.cvc [hunk 3]
```diff
@@ -47,48 +47,39 @@ endif
...
-        $(SIM_BUILD)/sim.vvp $(SIM_ARGS) $(EXTRA_ARGS) $(PLUSARGS)
-endif
+
+# Compilation phase
+$(SIM_BUILD)/cvcsim: $(SIM_BUILD) $(VERILOG_SOURCES) $(CUSTOM_COMPILE_DEPS)
+	$(CMD) +verbose -o $(SIM_BUILD)/cvcsim $(CVC_ARGS) +loadvpi=$(LIB_DIR)/libcocotbvpi_modelsim:vlog_startup_routines_bootstrap $(COMPILE_ARGS) $(EXTRA_ARGS) $(VERILOG_SOURCES)
 
 # Execution phase
-ifeq ($(CVC_ITERP),1)
-    debug:  $(CUSTOM_SIM_DEPS)
```
Comment (eric-wieser) [inline cocotb/share/makefiles/simulators/Makefile.cvc:80]: No more debug target?

File: cocotb/share/makefiles/simulators/Makefile.cvc [hunk 4]
```diff
@@ -47,48 +47,39 @@ endif
...
 else
-    $(COCOTB_RESULTS_FILE): $(SIM_BUILD)/sim.vvp $(CUSTOM_SIM_DEPS)
-	MODULE=$(MODULE) \
-        TESTCASE=$(TESTCASE) TOPLEVEL=$(TOPLEVEL) TOPLEVEL_LANG=$(TOPLEVEL_LANG) COCOTB_SIM=1 \
-        $(SIM_BUILD)/sim.vvp $(SIM_ARGS) $(EXTRA_ARGS) $(PLUSARGS)
-endif
+
+# Compilation phase
+$(SIM_BUILD)/cvcsim: $(SIM_BUILD) $(VERILOG_SOURCES) $(CUSTOM_COMPILE_DEPS)
+	$(CMD) +verbose -o $(SIM_BUILD)/cvcsim $(CVC_ARGS) +loadvpi=$(LIB_DIR)/libcocotbvpi_modelsim:vlog_startup_routines_bootstrap $(COMPILE_ARGS) $(EXTRA_ARGS) $(VERILOG_SOURCES)
```
Comment (eric-wieser) [inline cocotb/share/makefiles/simulators/Makefile.cvc:68]: Any idea why we use the modelsim one?

File: cocotb/share/makefiles/simulators/Makefile.cvc (hunk 3 above)
Comment (themperek) [reply cocotb/share/makefiles/simulators/Makefile.cvc:80]: I think if needed one can now easily highjack the makefile and modify. We should keep Makefiles simple. But this is just opinion.

File: cocotb/share/makefiles/simulators/Makefile.cvc (hunk 4 above)
Comment (themperek) [reply cocotb/share/makefiles/simulators/Makefile.cvc:68]: CVC is not very popular and at some point needed same ifdefs as modelsim. Not sure if need to make another library just for CV .

File: cocotb/share/makefiles/simulators/Makefile.cvc (hunk 3 above)
Comment (themperek) [reply cocotb/share/makefiles/simulators/Makefile.cvc:80]: Do you think I should change this?

File: cocotb/share/makefiles/simulators/Makefile.cvc (hunk 4 above)
Comment (themperek) [reply cocotb/share/makefiles/simulators/Makefile.cvc:68]: Do you think I should change this?

File: cocotb/share/makefiles/simulators/Makefile.cvc (hunk 3 above)
Comment (eric-wieser) [reply cocotb/share/makefiles/simulators/Makefile.cvc:80]: Do we have `debug` in the other makefiles? Seems like we should be consistent.

File: cocotb/share/makefiles/simulators/Makefile.cvc (hunk 4 above)
Comment (eric-wieser) [reply cocotb/share/makefiles/simulators/Makefile.cvc:68]: Ah, found what I was looking for in some removed code:

https://github.com/cocotb/cocotb/blob/b30cf07cc81a650b71548ce3e24d0820b6f7d802/cocotb/share/makefiles/Makefile.inc#L58-L61

I think a single comment here explaining that this is intentional would do.

File: cocotb/share/makefiles/simulators/Makefile.cvc (hunk 3 above)
Comment (themperek) [reply cocotb/share/makefiles/simulators/Makefile.cvc:80]: It looks only `icarus` and `ventilator` have `debug`. Consider CVC is not popular.

Comment (eric-wieser) [reply cocotb/share/makefiles/simulators/Makefile.cvc:80]: That's a good argument.

File: documentation/source/simulator_support.rst [hunk 5]
```diff
@@ -276,4 +276,4 @@ set :make:var:`SIM` to ``cvc``:
 
     make SIM=cvc
 
-Note that cocotb's makefile is using CVC's interpreted mode.
+To enable interpreted mode, use ``EXTRA_ARGS=+interp make ...``.
```
Comment (cmarqu) [inline documentation/source/simulator_support.rst:279]: Maybe quotes here would make it clearer that this isn't just a typo for ``+=``:

```suggestion
To enable interpreted mode, use ``EXTRA_ARGS="+interp" make ...``.
```

**Comment by eric-wieser** [comment]
Needs a rebase

**Comment by themperek** [comment]
This is outdated so I close.

**Pull request closed without merging**